app/scripts/controllers/locationPickerMap.ctrl.js

```diff
@@ -1,57 +1,192 @@
+    color: 'blue',
+    fillColor:'',
+    dashArray: '',
+    fillOpacity: 0
   };
+  function highlightFeature(e) {
+    if (!ctrl.selectedMunicipalitiesObj.idInArray(e.target.feature.id)) {
+      var layer = e.target;
+      layer.setStyle(outlineStyle);
 
-  function onMapClick(map, evt) {
-    ctrl.selectedMunicipalities = [];
-
-    var selectionQuery = new esri.tasks.Query();
-    var tol = map.extent.getWidth()/map.width * 5;
-    var x = evt.mapPoint.x;
-    var y = evt.mapPoint.y;
-    var queryExtent = new esri.geometry.Extent(x - tol, y - tol, x + tol, y + tol, evt.mapPoint.spatialReference);
-    selectionQuery.geometry = queryExtent;
-    var layer = map.getLayer(map.graphicsLayerIds[0]);
-    layer.queryFeatures(selectionQuery, function(queryResult){
-      queryResult.features.forEach(function(feature){
-        if (feature.attributes && feature.attributes.MUNICIPALITY) {
-          municipalities.every(function (realMunicip) {
-            if (realMunicip.municipality_name.toLowerCase() === feature.attributes.MUNICIPALITY.toLowerCase()) {
-              ctrl.selectedMunicipalities.push(realMunicip);
-              return false;
-            }
-
-            return true;
-          });
-        } else {
-          console.error('Received a map feature with the incorrect attributes. Something is wrong with the ESRI configuration.');
-        }
-      });
-    });
+      if (!L.Browser.ie && !L.Browser.opera && !L.Browser.edge) {
+        layer.bringToFront();
+      }
+    }
+    ctrl.mousedOverMunicipality = muniObj(e.target.feature.id,e.target.feature.properties.municipality);
+  }
 
-    $scope.$apply();
+  function resetHighlight(e) {
+    if (!ctrl.selectedMunicipalitiesObj.idInArray(e.target.feature.id)) {
+      geoJson.resetStyle(e.target);
+    }
+    ctrl.mousedOverMunicipality = "";
+  }
+
+  function selectMunicipality(e) {
+    if (ctrl.selectedMunicipalitiesObj.municipalityClicked(muniObj(e.target.feature.id,e.target.feature.properties.municipality))){
+      var layer = e.target;
+      layer.setStyle(highlightStyle);
+      if (!L.Browser.ie && !L.Browser.opera && !L.Browser.edge) {
+        layer.bringToFront();
+      }
+    }else{
+      geoJson.resetStyle(e.target);
+    }
   }
 
-  esriRegistry.get('stlMap').then(function(map){
-    map.on('click', function(e) {
-      onMapClick(map, e);
+  function onEachFeature(feature, layer) {
+    layer.on({
+      mouseover: highlightFeature,
+      mouseout: resetHighlight,
+      click:selectMunicipality
     });
+  }
+
+  leafletData.getMap("municipalityMap").then(function(map){
+    geoJson = L.geoJSON(countyMunicipalityBoundaryData,{
+      style: function(feature) {
+        return {
+          fillColor: "",
+          weight: 1,
+          opacity: 1,
+          color: 'blue',
+          dashArray: '',
+          fillOpacity: 0
+        }
+      },
+      onEachFeature: onEachFeature
+    }).addTo(map);
+
   });
 
+
+  function muniObj(muniId, muniName){
```

Comment:
If you need to make a new object (muniObj is probably not the name you want), please define it in a new file. Probably in a "models" or "domain" folder. And I would just call it Municipality.